test-data/unit/check-overloading.test

```diff
@@ -6474,7 +6474,7 @@ eggs = lambda: 'eggs'
 reveal_type(func(eggs))  # N: Revealed type is "def (builtins.str) -> builtins.str"
 
 spam: Callable[..., str] = lambda x, y: 'baz'
-reveal_type(func(spam))  # N: Revealed type is "def (*Any, **Any) -> builtins.str"
+reveal_type(func(spam))  # N: Revealed type is "def (*Any, **Any) -> Any"
```

Comment:
I'm surprised about this change (just purely looking at output): mypy seems to think `spam` is going to be using a single overload of `func`, and both return the return type unchanged. How is `str` turning into `Any`...?

Reply:
This one I understand, there is a rule for overloads that if
* arguments match more than one overload variant
* the multiple match is caused by `Any` types in argument types
* return types in matched overloads are not all equivalent

then the inferred return type becomes `Any`. IIUC all three conditions are satisfied in this case, so it is probably OK.